### PR TITLE
Fix show more focus behaviour

### LINKED ISSUE #6343: Fix focus behaviour for `ShowMore` container
Currently, when a user clicks the 'Show More' button on a Fronts container to load a new set of cards, their focus will remain on the button and the cards will be loaded _above_ the button in the DOM.

The new cards should be announced to screen readers. But for screen readers and others who are navigating using the `tab` key, it may be surprising that they have to tab _backwards_ to get to the new cards.

---

(prior to clicking the 'More' button)

---

(after clicking the 'More' button; new cards are loaded above it in the DOM)

---

**Potential solutions:**

1. Programmatically move focus to the new cards when they are loaded in.
2. Use CSS to place the button _visually_ below the cards, while keeping it above them in the DOM, so that you could tab forwards from the button to the cards.
3. Place the button above the cards both visually and in the DOM.

Option (1) currently seems to be the preferred route, but this is the topic of continuing discussion with UX and design.

### PR DESCRIPTION
**Pull request by bryophyta**
## What does this change?

Programmatically focuses the first new card in the 'show more' section, when that section is loaded.

## Why?

Because the new cards are rendered above the button itself. So with default tab behaviour keyboard users will need to tab backwards through the new cards to get to the beginning of the list. And it won't be obvious to screen reader users where they can find the new content.

Programmatically shifting focus is generally discouraged, but afaict is permissible when it's done in response to user interaction (here, clicking the 'show more' button). In addition, an aria description is added to the button to signal the behaviour to screen reader users.

Closes #6343

### CODE REVIEW

**Comment by bryophyta** [comment]
@isabellapurchas if you'd like to test this out at some point then let me know and we can do a deployment to CODE at some point?

File: dotcom-rendering/src/web/components/ShowMore.importable.tsx [hunk 1]
```diff
@@ -88,6 +84,22 @@ export const ShowMore = ({
 
 	const showMoreContainerId = `show-more-${containerId}`;
 
+	useEffect(() => {
+		/**
+		 * Focus the first of the new cards when they're loaded in.
+		 * There's no need to check `isOpen` here because if `isOpen` is
+		 * `false` then `filteredData` will be `undefined`.
+		 * */
+		if (filteredData && filteredData.length > 0) {
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/ShowMore.importable.tsx:93]: Might be more confusing, but 0 is falsy, so you could do:

```suggestion
		if (filteredData?.length) {
```

**Review by mxdvl — APPROVED** [review]
That looks lovely!

File: dotcom-rendering/src/web/components/ShowMore.importable.tsx (hunk 1 above)
Comment (bryophyta) [reply dotcom-rendering/src/web/components/ShowMore.importable.tsx]: yeah that's a good call I think -- I'd been thinking that it was good to be more explicit, but I think we use optional chaining widely enough in the codebase now that this should be fine 👍

**Pull request merged**